Bigtable/src/Instance.php

```diff
@@ -0,0 +1,271 @@
+        array $lroCallables,
+        $projectId,
+        $instanceId,
+        array $info = []
+    ) {
+        $this->connection = $connection;
+        $this->projectId = $projectId;
+        $this->name = $this->fullyQualifiedInstanceName($instanceId, $projectId);
+        $this->info = $info;
+        $this->setLroProperties($lroConnection, $lroCallables, $this->name);
+    }
+
+    /**
+     * Return the instance name.
+     *
+     * Example:
+     * ```
+     * $name = $instance->name();
+     * ```
+     *
+     * @return string
+     */
+    public function name()
+    {
+        return $this->name;
+    }
+
+    /**
+     * Return the service representation of the instance.
+     *
+     * This method may require a service call.
+     *
+     *
+     * @param array $options [optional] Configuration options.
+     */
+    public function info(array $options = [])
+    {
+        throw new \BadMethodCallException('This method is not implemented yet');
+    }
+
+    /**
+     * Check if the instance exists.
+     *
+     * This method requires a service call.
+     *
+     *
+     * @param array $options [optional] Configuration options.
+     */
+    public function exists(array $options = [])
+    {
+        throw new \BadMethodCallException('This method is not implemented yet');
+    }
+
+    /**
+     * Fetch a fresh representation of the instance from the service.
+     *
+     *
+     * @codingStandardsIgnoreStart
+     * @see https://cloud.google.com/bigtable/docs/reference/admin/rpc/google.bigtable.admin.v2#google.bigtable.admin.v2.GetInstanceRequest GetInstanceRequest
+     * @see https://cloud.google.com/bigtable/docs/reference/admin/rpc/google.bigtable.admin.v2#instance Instance
+     * @codingStandardsIgnoreEnd
+     *
+     * @param array $options [optional] Configuration options.
+     */
+    public function reload(array $options = [])
+    {
+        throw new \BadMethodCallException('This method is not implemented yet');
+    }
+
+    /**
+     * Return the instance state.
+     *
+     * When instances are created or updated, they may take some time before
+     * they are ready for use. This method allows for checking whether an
+     * instance is ready.
+     *
+     *
+     * @param array $options [optional] Configuration options.
+     */
+    public function state(array $options = [])
+    {
+        throw new \BadMethodCallException('This method is not implemented yet');
+    }
+
+    /**
+     * Create a new instance.
+     *
+     * Example:
+     * ```
+     * use Google\Cloud\Bigtable\BigtableClient;
+     * $bigtable = new BigtableClient();
+     * $instanceType = Google\Cloud\Bigtable\Instance::INSTANCE_TYPE_DEVELOPMENT;
+     * $operation = $instance->create(
+     *     'instanceId',
+     *     $instanceType,
+     *     ['foo' => 'bar'],
+     *     [$bigtable->clusterMetadata('my-cluster', 'us-east1-b')]
+     * );
+     * ```
+     *
+     * @codingStandardsIgnoreStart
```

Comment:
Just curious, what standard do we ignore here?

Reply:
Looks like line length